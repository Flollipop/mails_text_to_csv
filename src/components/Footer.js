import React from 'react';

const repoLink = 'https://github.com/Flollipop/mails_text_to_csv';
const footerLabel = 'See the code';

export function Footer() {
  return (
    <footer className={'App-footer'}>
      <a className={'App-link'} href={repoLink} target='_blank' rel='noopener noreferrer'>
        {footerLabel}
      </a>
    </footer>
  );
}
