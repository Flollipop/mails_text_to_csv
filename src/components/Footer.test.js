import React from 'react';
import ReactDOM from 'react-dom';
import { Footer } from './Footer';
import { render } from '@testing-library/react';

test('render footer - smoke test', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Footer />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('github link', () => {
  const { getByText } = render(<Footer />);
  const linkElement = getByText('See the code');

  expect(linkElement.href).toBe('https://github.com/Flollipop/mails_text_to_csv');
  expect(linkElement).toBeInTheDocument();
});
