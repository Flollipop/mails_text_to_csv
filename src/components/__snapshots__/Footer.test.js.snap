// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot 1`] = `
<footer
  className="App-footer"
>
  <a
    className="App-link"
    href="https://github.com/Flollipop/mails_text_to_csv"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      className="Github-logo"
    >
      github-logo.svg
    </svg>
    See the code
  </a>
</footer>
`;
